feat: Implement Outlook Calendar Cache with Microsoft Graph Change No…

harshith-vaddiparthy opened the pull request:
…tifications

- Added caching support to Office365CalendarService
- Created MicrosoftGraphSubscriptionService for webhook management
- Added webhook endpoint for Microsoft Graph notifications
- Updated schema to support Microsoft Graph subscriptions
- Added tests for the implementation

Fixes #21050

## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Visual Demo (For contributors especially)

A visual demonstration is strongly recommended, for both the original and new change **(video / image - any one)**.


## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Added caching for Outlook/Office 365 calendar availability and real-time updates using Microsoft Graph change notifications. This reduces API calls and keeps calendar data fresh.

- **New Features**
  - Cached busy times for Outlook calendars to improve performance.
  - Webhook endpoint and subscription service for Microsoft Graph notifications.
  - Automatic cache invalidation and refresh when calendar events change.
  - Updated database schema to store Microsoft Graph subscription info.

<!-- End of auto-generated description by cubic. -->

harshith-vaddiparthy committed:
feat: Implement Outlook Calendar Cache with Microsoft Graph Change Notifications

- Added caching support to Office365CalendarService
- Created MicrosoftGraphSubscriptionService for webhook management
- Added webhook endpoint for Microsoft Graph notifications
- Updated schema to support Microsoft Graph subscriptions
- Added tests for the implementation

Fixes #21050

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=22120) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=22120) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=22120) it.</sub>

harshith-vaddiparthy committed:
fix: Address review comments from cubic-dev-ai

- Remove unused eventTypeId field from selectedCalendar query
- Remove sensitive validationToken from debug logs
- Fix TypeScript any type warning

kart1ka commented:
Type checks and tests seem to be failing. Can you pls address them?

harshith-vaddiparthy committed:
Merge branch 'calcom:main' into main
